service/model_manager/model_installer.py

```diff
@@ -56,32 +56,29 @@ def parse_wget_output(output):
     match = re.search(r'\d+%|\d+K \d+Kb/s', output)
     return match.group(0) if match else ""
 
-import queue
-import threading
-
-# Create a Queue object
-download_queue = queue.Queue()
+# Create a dictionary to store download tasks
+download_tasks = []
+# Create a lock object to synchronize access to the dictionary
+download_tasks_lock = threading.Lock()
 # Create a thread that will execute the download tasks
 download_thread = None
-# Create a Condition object to share the download progress
-download_progress = threading.Condition()
-download_progress.value = {'progress': "", 'error': None}
 
 def download_worker():
-    while not download_queue.empty():
-        # Get a download task from the queue
-        (url, save_path) = download_queue.get()
-
-        # Execute the download task and update the download progress
-        download_url_with_agent(url, save_path)
-
-        # Indicate that the task is done
-        download_queue.task_done()
-
-    # If all tasks are done, reset the progress and exit
-    with download_progress:
-        download_progress.value['progress'] = ""
-        download_progress.notify_all()
+    while True:
```

Comment:
i wonder this compared to `while not download_tasks.empty()`

Reply:
`Queue` is FIFO, so it's impossible to show the queue to the user, so I had to switch to `List`, which doesn't have `empty()` method